flashinfer/kda_kernels/kda_decode_bf16_state.py

```
@@ -0,0 +1,1367 @@
+        assert A_log is not None, "A_log is required when use_gate_in_kernel=True"
+        assert A_log.dtype == torch.float32, f"A_log must be float32, got {A_log.dtype}"
+    if lower_bound is not None:
+        assert use_gate_in_kernel, "lower_bound requires use_gate_in_kernel=True"
+
+    # --- Prepare cu_seqlens / state ---
+    if cu_seqlens is not None:
+        if B != 1:
+            raise ValueError(
+                f"Batch size must be 1 when using cu_seqlens, got B={B}. "
+                f"Flatten variable-length inputs before processing."
+            )
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)